test/jdk/java/awt/print/PrinterJob/PrintNullString.java

```diff
@@ -21,308 +21,203 @@
+    }
 
- public void actionPerformed(ActionEvent e) {
+    public void actionPerformed(ActionEvent e) {
 
-   PrinterJob pj = PrinterJob.getPrinterJob();
+        PrinterJob pj = PrinterJob.getPrinterJob();
 
-   if (pj != null && pj.printDialog()) {
+        if (pj != null && pj.printDialog()) {
 
-       pj.setPrintable(c);
-       try {
-            pj.print();
-      } catch (PrinterException pe) {
-      } finally {
-         System.err.println("PRINT RETURNED");
-      }
-   }
- }
+            pj.setPrintable(c);
+            try {
+                pj.print();
+            } catch (PrinterException pe) {
+            } finally {
+                System.out.println("PRINT RETURNED");
+            }
+        }
+    }
 
- class TextCanvas extends Panel implements Printable {
+    static class TextCanvas extends Panel implements Printable {
 
-    String nullStr = null;
-    String emptyStr = new String();
-    AttributedString nullAttStr = null;
-    AttributedString emptyAttStr = new AttributedString(emptyStr);
-    AttributedCharacterIterator nullIterator = null;
-    AttributedCharacterIterator emptyIterator = emptyAttStr.getIterator();
+        String nullStr = null;
+        String emptyStr = "";
+        AttributedString emptyAttStr = new AttributedString(emptyStr);
+        AttributedCharacterIterator nullIterator = null;
+        AttributedCharacterIterator emptyIterator = emptyAttStr.getIterator();
 
-    public int print(Graphics g, PageFormat pgFmt, int pgIndex) {
+        public int print(Graphics g, PageFormat pgFmt, int pgIndex) {
 
-      if (pgIndex > 0)
-         return Printable.NO_SUCH_PAGE;
+            if (pgIndex > 0)
+                return Printable.NO_SUCH_PAGE;
 
-      Graphics2D g2d = (Graphics2D)g;
-      g2d.translate(pgFmt.getImageableX(), pgFmt.getImageableY());
+            Graphics2D g2d = (Graphics2D) g;
+            g2d.translate(pgFmt.getImageableX(), pgFmt.getImageableY());
 
-      paint(g);
+            paint(g);
 
-      return Printable.PAGE_EXISTS;
-    }
+            return Printable.PAGE_EXISTS;
+        }
 
-    public void paint(Graphics g1) {
-        Graphics2D g = (Graphics2D)g1;
+        public void paint(Graphics g1) {
+            Graphics2D g = (Graphics2D) g1;
 
-        // API 1: null & empty drawString(String, int, int);
-        try {
-             g.drawString(nullStr, 20, 40);
-             g.drawString("FAILURE: No NPE for null String, int", 20, 40);
-        } catch (NullPointerException e) {
-          g.drawString("caught expected NPE for null String, int", 20, 40);
-        }/* catch (Exception e) {
+            // API 1: null & empty drawString(String, int, int);
+            try {
+                g.drawString(nullStr, 20, 40);
+                g.drawString("FAILURE: No NPE for null String, int", 20, 40);
+            } catch (NullPointerException e) {
+                g.drawString("caught expected NPE for null String, int", 20, 40);
+            }/* catch (Exception e) {
```

Comment:
I guess u can remove these instead of commenting?

Reply:
These lines are commented in existing code. Felt it may be useful for further debugging in case of failure, so kept as it is.

Reply:
Ok, else the fix is fine for me.